### PR TITLE
Adds #include to address build error

### PR DESCRIPTION
**Pull request by geoffthemedio**
https://github.com/freeorion/freeorion/issues/2308#issuecomment-441685768

### CODE REVIEW

File: combat/CombatEvents.h [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 #include <map>
 #include <memory>
+#include <string>
```
Comment (jbeich) [inline combat/CombatEvents.h]: [Not enough](https://ptpb.pw/orl0). Maybe move to `combat/CombatEvent.h` (singular), included by this file on line 16. Both use `std::string` but  `combat/CombatEvents.h` (plural) already bootlegs `<vector>` via `combat/CombatEvent.h`.

Comment (jbeich) [reply combat/CombatEvents.h]: Also sort includes e.g., `<set>` before `<string>` before `<vector>`.

Comment (geoffthemedio) [reply combat/CombatEvents.h]: now?

**Review by jbeich — COMMENTED** [review]
Thanks. [Builds fine](https://ptpb.pw/Eu09) now.

File: combat/CombatEvent.h [hunk 2]
```diff
@@ -9,6 +9,8 @@
 #include "../util/Export.h"
 
 #include <memory>
+#include <set>
```
[the remark this answers is not in the document]
Comment (geoffthemedio) [reply combat/CombatEvent.h]: Can you clearly and specifically indicate just which changes are needed then? Or make your own pull request?

Alternatively, I think you should be able to propose changes to this pull request using the GitHub web interface.

[the remark this answers is not in the document]
Comment (jbeich) [reply combat/CombatEvent.h]: Sorry for the confusion. As I've said in https://github.com/freeorion/freeorion/issues/2308#issuecomment-441685768 only `<string>` in this file is required to unbreak build with Boost 1.69 (again). Whether to explicitly add other includes to maintain a particular style or header self-sufficiency is up to the project.

[the remark this answers is not in the document]
Comment (jbeich) [reply combat/CombatEvent.h]: Yes. Thanks.

**Pull request merged**